File: server/src/main/java/com/aifriend/reminder/application/ReminderSchedulerPort.java
package com.aifriend.reminder.application;

import java.time.Instant;

/**
 * 个人提醒调度端口。
 *
 * <p>当前仅占位，不注册实现、不开放 API，也不能自动触发微信外发。
 *
 * @author Codex
 * @since 1.0.0
 */
public interface ReminderSchedulerPort {

    /**
     * 安排本地或服务端提醒。
     *
     * @param reminderId 提醒标识
     * @param triggerAt UTC 触发时间
     */
    void schedule(String reminderId, Instant triggerAt);
}

File: android/app/src/main/java/com/aifriend/core/reminder/ReminderSchedulerPort.kt
package com.aifriend.core.reminder

import java.time.Instant

/**
 * 本地提醒调度端口。
 *
 * 当前只占位，不注册实现、不显示 UI，也不能自动触发微信外发。
 */
interface ReminderSchedulerPort {
    /** 安排一个使用 UTC 时间的提醒。 */
    suspend fun schedule(reminderId: String, triggerAt: Instant)
}
